src/simularium/LocalFileSimulator.ts

```diff
@@ -157,20 +159,16 @@ export class LocalFileSimulator implements ISimulator {
     }
 
     public gotoRemoteSimulationTime(timeNs: number): void {
-        for (
-            let frame = 0,
-                numFrames = this.simulariumFile.spatialData.bundleData.length;
-            frame < numFrames;
-            frame++
-        ) {
-            const frameTime = this.simulariumFile.spatialData.bundleData[frame]
-                .time;
-            if (timeNs <= frameTime) {
-                const theFrameNumber = Math.max(frame, 0);
-                this.onTrajectoryDataArrive(this.getFrame(theFrameNumber));
-                break;
-            }
-        }
+        const { bundleData } = this.simulariumFile.spatialData;
+        const { timeStepSize } = this.simulariumFile.trajectoryInfo;
+
+        // Find the index of the frame that has the time matching our target time
+        const frameNumber = bundleData.findIndex((bundleData) => {
+            return compareFloats(bundleData.time, timeNs, timeStepSize) === 0;
+        });
+
+        // frameNumber is -1 if findIndex() above doesn't find a match
+        this.onTrajectoryDataArrive(this.getFrame(Math.max(frameNumber, 0)));
```

Comment:
slight behavior change.  the old code never calls this function if it fails to find a time that satisfies the criteria

Reply:
Do you want to return frame zero if the time passed in is out of range, or do nothing?  That's the behavior change I am seeing.  In the old code, it was possible to loop thru all values and never hit the condition, in which case nothing happens.  In the new code, it ALWAYS calls onTrajectoryDataArrive.

Reply:
I think you're right Dan, I will modify this (and VisData) to preserve the previous behavior in that regard

Reply:
I'm not sure if there is a point to `Math.max(frameNumber, 0))` anymore though. `frameNumber` should always be an integer greater than or equal to 0, or be -1 in the case `findIndex()` doesn't find a match